tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py

```diff
@@ -2,152 +2,277 @@
 import logging
 from pathlib import Path
 
-import pytest
-
 from cg.cli.workflow.balsamic.base import config_case
 
 EXIT_SUCCESS = 0
 
 
 def test_without_options(cli_runner, balsamic_context):
-    """Test command with dry option"""
-
-    # GIVEN
-
+    """Test command without case_id"""
+    # GIVEN NO case_id
     # WHEN dry running without anything specified
     result = cli_runner.invoke(config_case, obj=balsamic_context)
-
     # THEN command should mention argument
     assert result.exit_code != EXIT_SUCCESS
     assert "Missing argument" in result.output
 
 
 def test_with_missing_case(cli_runner, balsamic_context, caplog):
-    """Test command with case to start with"""
-
-    # GIVEN case-id not in database
+    """Test command with invalid case to start with"""
+    caplog.set_level(logging.WARNING)
+    # GIVEN case_id not in database
     case_id = "soberelephant"
-    caplog.set_level(logging.ERROR)
-
+    assert not balsamic_context["BalsamicAnalysisAPI"].store.family(case_id)
     # WHEN running
     result = cli_runner.invoke(config_case, [case_id], obj=balsamic_context)
-
-    # THEN command should successfully call the command it creates
+    # THEN command should NOT successfully call the command it creates
     assert result.exit_code != EXIT_SUCCESS
+    # THEN ERROR log should be printed containing invalid case_id
+    assert case_id in caplog.text
 
-    with caplog.at_level(logging.ERROR):
-        assert case_id in caplog.text
-
-
-def test_dry(cli_runner, balsamic_context, balsamic_case, caplog):
-    """Test command with --dry option"""
 
+def test_without_samples(cli_runner, balsamic_context, caplog):
+    """Test command with case_id and no samples"""
+    caplog.set_level(logging.WARNING)
     # GIVEN case-id
-    case_id = balsamic_case.internal_id
-    caplog.set_level(logging.INFO)
-
+    case_id = "no_sample_case"
     # WHEN dry running with dry specified
     result = cli_runner.invoke(config_case, [case_id, "--dry-run"], obj=balsamic_context)
     # THEN command should print the balsamic command-string
-    assert result.exit_code == EXIT_SUCCESS
-    with caplog.at_level(logging.INFO):
-        assert "balsamic" in caplog.text
-        assert case_id in caplog.text
-
+    assert result.exit_code != EXIT_SUCCESS
+    # THEN warning should be printed that no config file is found
+    assert case_id in caplog.text
+    assert "0" in caplog.text
```

Comment:
why "0"?

Reply:
Taken error message from log : number of tumor samples cannot be 0 to run balsamic
But I will add the whole string because thinking about it, there are other ways 0 can end up in log